shared/src/commonMain/kotlin/com/kcc/kmmhackathon/shared/network/MealApi.kt

```diff
@@ -32,18 +33,19 @@ class MealApi {
 
     suspend fun getSortedMeals(userLat: Double, userLon: Double, distanceUnit: DistanceUnit): List<Meal> {
         var meals: List<Meal> = getAllMeals()
-        for (meal in meals) {
-            meal.distance = locationUtil.getDistance(
+        meals.filter { SharedDate().isBefore(SharedDate(it.expiryDate.toLong())) }
+        meals.forEach {
+            it.distance = locationUtil.getDistance(
                 userLat,
                 userLon,
-                meal.locationLat.toDouble(),
-                meal.locationLong.toDouble(),
+                it.locationLat.toDouble(),
+                it.locationLong.toDouble(),
                 distanceUnit
             )
+            it.expiryDate = dateFormattingUtil.convertTimeStamp(it.expiryDate.toLong())
+            it.availableFromDate = dateFormattingUtil.convertTimeStamp(it.availableFromDate.toLong())
         }
-        return meals
-            .filter { SharedDate().isBefore(SharedDate(it.expiryDate.toLong())) }
-            .sortedWith(compareBy { it.distance })
+        return meals.sortedBy { it.distance }
     }
```

Comment:
There might be a nicer way to write this and it's a bit difficult to read with the removed lines so:
```
        var meals: List<Meal> = getAllMeals()
        meals.filter { SharedDate().isBefore(SharedDate(it.expiryDate.toLong())) }
        meals.forEach {
            it.distance = locationUtil.getDistance(
                userLat,
                userLon,
                meal.locationLat.toDouble(),
                meal.locationLong.toDouble(),
                distanceUnit
            )
            it.expiryDate = dateFormattingUtil.convertTimeStamp(it.expiryDate.toLong())
            it.availableFromDate = dateFormattingUtil.convertTimeStamp(it.availableFromDate.toLong())
        }
        return meals.sortedBy { it.distance }
```

I was expecting I could just link those calls together 
`.filter{ }.forEach{ }.sortedBy{ }`

but ran into a few problems. If that's because i've done something dumb please let me know 😆

Reply:
Nice that you added readable code here 👍